src/XR.tsx

```diff
@@ -159,17 +159,29 @@ function XRCanvas({ children, ...rest }: ContainerProps) {
   )
 }
 
-export function VRCanvas({ children, ...rest }: ContainerProps) {
+export function VRCanvas({ onCreated, children, ...rest }: ContainerProps) {
   return (
-    <XRCanvas onCreated={({ gl }) => void document.body.appendChild(VRButton.createButton(gl))} {...rest}>
+    <XRCanvas
+      onCreated={(state) => {
+        onCreated?.(state)
+
+        document.body.appendChild(VRButton.createButton(state.gl))
+      }}
+      {...rest}>
       {children}
     </XRCanvas>
   )
 }
 
-export function ARCanvas({ children, sessionInit, ...rest }: ContainerProps & { sessionInit?: any }) {
+export function ARCanvas({ onCreated, children, sessionInit, ...rest }: ContainerProps & { sessionInit?: any }) {
   return (
-    <XRCanvas onCreated={({ gl }) => void document.body.appendChild(ARButton.createButton(gl, sessionInit))} {...rest}>
+    <XRCanvas
+      onCreated={(state) => {
+        onCreated?.(state)
+
+        document.body.appendChild(ARButton.createButton(state.gl, sessionInit))
```

Comment:
Should we allow `sessionInit` to be specified for `XRCanvas`?

Reply:
Doesn't seem so: https://github.com/pmndrs/react-xr/blob/master/src/webxr/VRButton.js#L5.